### ISSUE TITLE
SC2060: Incorrect warning with tr.

### ISSUE THREAD

**Issue by orbea** [koalaman/shellcheck#1428]
#### For bugs
- Rule Id: `SC2060`
- My shellcheck version: `online`
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

FOO=foo
echo $FOO | tr [:lower:] [:upper:]

```

#### Here's what shellcheck currently says:

```
Line 4:
echo $FOO | tr [:lower:] [:upper:]
               ^-- SC2060: Quote parameters to tr to prevent glob expansion.
                         ^-- SC2060: Quote parameters to tr to prevent glob expansion.
```

#### Here's what I wanted or expected to see:

Nothing.

The variable `$FOO` will never suffer from word splitting, `echo` does not suffer from such warnings and neither should `tr`.

**Comment by Gandalf-** [comment]
Relevant snippet

https://github.com/koalaman/shellcheck/blob/master/src/ShellCheck/Checks/Commands.hs#L149-L150

**Comment by Gandalf-** [comment]
Introduced originally 6 years ago in this commit

https://github.com/koalaman/shellcheck/commit/b718e5f10857108ab5de7aa13681ef830b5d2c66

**Comment by matthewpersico** [comment]
It’s not complaining about $FOO; look at where the messages point; it’s complaining about the args to tr. Given that the args are constants, the check is incorrect , I believe.

**Comment by koalaman** [comment]
ShellCheck is correct. These arguments must be quoted to prevent them from being expanded as globs. 

Currently, it breaks in pretty random ways based on the contents of the current directory:

```
$ cat myscript
#!/bin/sh

FOO=foo
echo $FOO | tr [:lower:] [:upper:]

$ ./myscript
FOO

$ mkdir r && ./myscript
foo

$ touch l && ./myscript
tr: extra operand ‘r’
Try 'tr --help' for more information.
``

A correctly quoted version with `tr '[:lower:]' '[:upper:]'` will work correctly and output FOO in all the cases above, without being affected by unrelated files in the current directory.

**Comment by orbea** [comment]
Thanks for pointing this out, I don't understand how the unrelated files are even making a difference here?
